File: volatility3/app/schemas/image.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from ..models.image import ImageStatus


class ImageOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    filename: str
    size_bytes: int
    sha256: str
    os_hint: str | None
    os_version: str | None
    legacy_profile: str | None
    status: ImageStatus
    created_at: datetime

File: frontend/src/types/image.ts
export type ImageStatus = "uploading" | "identifying" | "ready" | "error";

export interface VolImage {
  id: string;
  filename: string;
  size_bytes: number;
  sha256: string;
  os_hint: string | null;
  os_version: string | null;
  legacy_profile: string | null;
  status: ImageStatus;
  created_at: string;
}
